charts/consul/templates/gateway-resources-configmap.yaml

```diff
@@ -66,6 +66,9 @@ data:
             {{- with .Values.meshGateway.hostNetwork }}
             hostNetwork: {{ . }}
             {{- end }}
+            {{- with .Values.meshGateway.dnsPolicy }}
+            dnsPolicy: {{ . }}
+            {{- end }}
```

Comment:
What does the `with` here do vs. something like
```yaml
{{- if .Values.meshGateway.dnsPolicy }}
dnsPolicy: {{ .Values.meshGateway.dnsPolicy }}
{{- end }}
```

Reply:
pretty much the same thing, it just allows you to reference the value with `.` rather than needing to reference the value again https://pkg.go.dev/text/template#hdr-Actions, quoted the relevant section here:
> {{with pipeline}} T1 {{end}}
	If the value of the pipeline is empty, no output is generated;
	otherwise, dot is set to the value of the pipeline and T1 is
	executed.

Reply:
ohhhhh I like that a lot.

TIL. Thanks for sharing!